### ISSUE TITLE
Loose connection with MySQL 

### ISSUE THREAD

**Issue by tristancaron** [prisma/photonjs#69]
When running a lot of operation at "the same time" against the database using a single instance of `Photon`, connections get refused.

Here a sample with `debug: true`
```
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +0ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +0ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +58ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +1ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +57ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +1ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +54ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +1ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +78ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +1ms
  engine request to http://localhost:52108/status failed, reason: connect ECONNREFUSED 127.0.0.1:52108 +54ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +0ms
  engine request to http://localhost:52107/status failed, reason: connect ECONNREFUSED 127.0.0.1:52107 +57ms
```

**Comment by matthewmueller** [comment]
Hi @tristancaron, do you mind sharing the code to reproduce that effect? Thanks!

**Comment by tristancaron** [comment]
I think it's related to the current limitation `Processes 1 request at a time, no parallelism during the early preview period.`

It's happening in jest when running tests. It runs parallel processes.

**Comment by schickling** [comment]
We'll pick this up as soon as the query engine (written in Rust) will support parallel queries. Probably in ~1 month from now.

**Comment by nlarusstone** [comment]
Hi,

Can we have an update on this? This is causing a "Too many connections issue" when we try to run out Jest tests.

Best,
Nicholas

**Comment by divyenduz** [comment]
@nlarusstone Do you have a reproduction? I was unable to reproduce this. 

I know what `now dev` might cause DB to overwhelm (as well as production lambda) because of the DB connection characteristics (irrespective of Photon). We have it documented here: https://github.com/prisma/prisma2/blob/master/docs/photon/deployment.md#photon-in-faas-environment-eg-aws-lambda-netlify

But a general connection leak is something I am unable to reproduce. 

Specifically about your environment: How many parallel connection can your handle? How many tests run in parallel?

**Comment by nlarusstone** [comment]
Sorry I should have updated here -- our issue was caused by having our jest tests fail and not exit appropriately, so the Photon connections just hung around and never exited.

We fixed this by killing the zombie connections and ensuring that Photon gets torn down no matter the type of exit

**Comment by divyenduz** [comment]
Awesome, thanks for reporting that, I would close this issue as it is not reproducible, we have documented this subject here: https://github.com/prisma/prisma2/blob/master/docs/photon/deployment.md#photon-in-faas-environment-eg-aws-lambda-netlify

If you think that this issue still exists, please open a new issue and tag me on that 🙌
